### PR TITLE
feat: support for Pekko Streams in the scala client

### LINKED ISSUE #660: Pekko support for scala client
Are there any plans to include support for Pekko Streams in the future?

### PR DESCRIPTION
**Pull request by utkuaydn**
Closes #660

## Proposed Changes

Support for Pekko Streams in the scala client

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `mvn test` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

**Review by bednar — CHANGES_REQUESTED** [review]
Hi @utkuaydn,

I want to express my sincere gratitude for your pull request. It's a substantial improvement to our client, and we truly appreciate the effort and expertise you've brought into this project. Given Lightbend's decision to license future Akka releases under the Business Software License — which is incompatible with Open Source usage — I agree with your perspective that use Pekko streams. 

From our point of view the Akka Streams should be replaced by Pekko Streams.

Could you please update your pull request to reflect these changes? 

1. Remove Akka Streams: Completely transition away from Akka Streams and implement Pekko streams throughout the client.
1. Upgrade Major Version: Update the major version of the client to 7.0.0 to reflect this significant change and its potential impacts on compatibility.

These updates will align our client with a fully open-source framework and ensure its usability and compliance with open-source licensing moving forward.

Once again, thank you for your contribution, and we look forward to seeing the updated pull request.

File: client-kotlin/README.md [hunk 1]
```diff
@@ -303,14 +303,14 @@ The latest version for Maven dependency:
 <dependency>
   <groupId>com.influxdb</groupId>
   <artifactId>influxdb-client-kotlin</artifactId>
-  <version>6.12.0</version>
+  <version>7.0.0</version>
```
Comment (bednar) [inline client-kotlin/README.md:306]: This should be actual stable version:
```suggestion
  <version>6.12.0</version>
```

File: client-kotlin/README.md [hunk 2]
```diff
@@ -303,14 +303,14 @@ The latest version for Maven dependency:
...
 </dependency>
 ```
   
 Or when using with Gradle:
 ```groovy
 dependencies {
-    implementation "com.influxdb:influxdb-client-kotlin:6.12.0"
+    implementation "com.influxdb:influxdb-client-kotlin:7.0.0"
```
Comment (bednar) [inline client-kotlin/README.md:313]: This should be actual stable version:
```suggestion
    implementation "com.influxdb:influxdb-client-kotlin:6.12.0"
```

File: client-legacy/README.md [hunk 3]
```diff
@@ -162,14 +162,14 @@ The latest version for Maven dependency:
 <dependency>
   <groupId>com.influxdb</groupId>
   <artifactId>influxdb-client-flux</artifactId>
-  <version>6.12.0</version>
+  <version>7.0.0</version>
```
Comment (bednar) [inline client-legacy/README.md:165]: This should be actual stable version:
```suggestion
  <version>6.12.0</version>
```

File: client-legacy/README.md [hunk 4]
```diff
@@ -162,14 +162,14 @@ The latest version for Maven dependency:
...
 </dependency>
 ```
   
 Or when using with Gradle:
 ```groovy
 dependencies {
-    implementation "com.influxdb:influxdb-client-flux:6.12.0"
+    implementation "com.influxdb:influxdb-client-flux:7.0.0"
```
Comment (bednar) [inline client-legacy/README.md:172]: This should be actual stable version:
```suggestion
    implementation "com.influxdb:influxdb-client-flux:6.12.0"
```

File: client-reactive/README.md [hunk 5]
```diff
@@ -433,14 +433,14 @@ The latest version for Maven dependency:
 <dependency>
   <groupId>com.influxdb</groupId>
   <artifactId>influxdb-client-reactive</artifactId>
-  <version>6.12.0</version>
+  <version>7.0.0</version>
```
Comment (bednar) [inline client-reactive/README.md:436]: This should be actual stable version:
```suggestion
  <version>6.12.0</version>
```

File: client-reactive/README.md [hunk 6]
```diff
@@ -433,14 +433,14 @@ The latest version for Maven dependency:
...
 </dependency>
 ```
   
 Or when using with Gradle:
 ```groovy
 dependencies {
-    implementation "com.influxdb:influxdb-client-reactive:6.12.0"
+    implementation "com.influxdb:influxdb-client-reactive:7.0.0"
```
Comment (bednar) [inline client-reactive/README.md:443]: This should be actual stable version:
```suggestion
    implementation "com.influxdb:influxdb-client-reactive:6.12.0"
```

File: client-scala/README.md [hunk 7]
```diff
@@ -257,14 +257,14 @@ The latest version for Maven dependency:
 <dependency>
   <groupId>com.influxdb</groupId>
   <artifactId>influxdb-client-scala_2.12</artifactId>
-  <version>6.12.0</version>
+  <version>7.0.0</version>
```
Comment (bednar) [inline client-scala/README.md:260]: This should be actual stable version:
```suggestion
  <version>6.12.0</version>
```

File: client-scala/README.md [hunk 8]
```diff
@@ -257,14 +257,14 @@ The latest version for Maven dependency:
...
 </dependency>
 ```
   
 Or when using with Gradle:
 ```groovy
 dependencies {
-    implementation "com.influxdb:influxdb-client-scala_2.12:6.12.0"
+    implementation "com.influxdb:influxdb-client-scala_2.12:7.0.0"
```
Comment (bednar) [inline client-scala/README.md:267]: This should be actual stable version:
```suggestion
    implementation "com.influxdb:influxdb-client-scala_2.12:6.12.0"
```

File: client-scala/README.md [hunk 9]
```diff
@@ -275,14 +275,14 @@ The latest version for Maven dependency:
 <dependency>
   <groupId>com.influxdb</groupId>
   <artifactId>influxdb-client-scala_2.13</artifactId>
-  <version>6.12.0</version>
+  <version>7.0.0</version>
```
Comment (bednar) [inline client-scala/README.md:278]: This should be actual stable version:
```suggestion
  <version>6.12.0</version>
```

File: client-scala/README.md [hunk 10]
```diff
@@ -275,14 +275,14 @@ The latest version for Maven dependency:
...
 </dependency>
 ```
   
 Or when using with Gradle:
 ```groovy
 dependencies {
-    implementation "com.influxdb:influxdb-client-scala_2.13:6.12.0"
+    implementation "com.influxdb:influxdb-client-scala_2.13:7.0.0"
```
Comment (bednar) [inline client-scala/README.md:285]: This should be actual stable version:
```suggestion
    implementation "com.influxdb:influxdb-client-scala_2.13:6.12.0"
```

File: spring/README.md [hunk 11]
```diff
@@ -99,13 +99,13 @@ The latest version for Maven dependency:
 <dependency>
   <groupId>com.influxdb</groupId>
   <artifactId>influxdb-spring</artifactId>
-  <version>6.12.0</version>
+  <version>7.0.0</version>
```
Comment (bednar) [inline spring/README.md:102]: This should be actual stable version:
```suggestion
  <version>6.12.0</version>
```

File: spring/README.md [hunk 12]
```diff
@@ -99,13 +99,13 @@ The latest version for Maven dependency:
...
 </dependency>
 ```
   
 Or when using with Gradle:
 ```groovy
 dependencies {
-    implementation "com.influxdb:influxdb-spring:6.12.0"
+    implementation "com.influxdb:influxdb-spring:7.0.0"
```
Comment (bednar) [inline spring/README.md:109]: This should be actual stable version:
```suggestion
    implementation "com.influxdb:influxdb-spring:6.12.0"
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,4 +1,8 @@
-## 6.13.0 [unreleased]
+## 7.0.0 [unreleased]
+
```
Comment (bednar) [inline CHANGELOG.md:2]: Please add clarification for replacement of the Akka Streams, something like:
```suggestion

:warning: This client version discontinues support for Akka Streams and introduces support for Pekko Streams instead. Apache Pekko is a fork of [Akka](https://github.com/akka/akka) 2.6.x, created after the Akka project adopted the Business Source License, which is not compatible with open-source usage.

```

**Review by bednar — CHANGES_REQUESTED** [review]
Thank you again for your PR! 👍 Before we can proceed with accepting it, there are a few requirements that need to be met:

1. Please rebase sources to fix conflict in `pom.xml`
2. Please satisfy our Checklist:
    image

and also:

**Comment by utkuaydn** [comment]
Hi @bednar,

I've completed the checklist and rebased the commit 👍

**Review by bednar — APPROVED** [review]
@utkuaydn thanks again for your PR 👍 

LGTM 🚀

**Comment by reyrobs** [comment]
Well done Utku!

**Pull request merged**